### DISCUSSION TITLE
iTunes library

### DISCUSSION THREAD

**Discussion by joea27** [music-assistant/support#3339] [Feature requests and ideas]
Can we get an option to use iTunes db or the iTunes itl file to organize our library and playlists. I have a music collection that is over 40,000 tracks. In iTunes it is organized in OCD type fashion and it would be great if that ability could be had.

**Comment by kfitzzz2222** [comment]
@jeanthom does your iTunes itl data parsing script still work? Wondering how you got the decryption key? I don’t see anything online about that anywhere

**Reply by jeanthom** [reply]
libitlp was more of a reverse engineering effort than a proper implementation of a file parser. At the time I published it, it had the best understanding of the itl format (yet far from complete), and https://home.vollink.com/gary/playlister/ituneslib.html got it further.

I would highly recommend against using it in production (it was my first "real" C project, it is lacking the safeguards that I would expect from a file parser, etc.). I you don't want to go down the rabbit hole of creating a new itl parser project, you can generate XML files of the library. It doesn't cover 100% of the .itl metadata but it might do the job.

Regarding the decryption key: soz mate there is no way I can help you on this one. I thought I made it clear on the README.

**Reply by kfitzzz2222** [reply]
No worries! Thank you very much for responding. I think I’m in a good enough spot to figure it out from here

**Reply by anline380** [reply]
@kfitzzz2222 Have you somehow managed to import the XML (with newer iTunes/Music versions, the XML file can, for example, be automatically generated on a Mac via the Music Library Exporter app) into Music Assistant? Many servers, such as Roon or LMS (Lyrion), support this feature.
